Documentation/AdvancedBuildInstructions.md

```diff
@@ -0,0 +1,74 @@
+# Advanced Build Instructions
+
+This file covers a few advanced scenarios that go beyond what the basic build guide provides.
+
+## Customizing the disk image
+
+To add, modify or remove files of the disk image's file system, e.g. to change the default keyboard layout, you can create a shell script with the name `sync-local.sh` in the project root, with content like this:
+
+```sh
+#!/bin/sh
+
+set -e
+
+cat << 'EOF' > mnt/etc/Keyboard.ini
+[Mapping]
+Keymap=de
+EOF
+```
+
+This will configure your keymap to German (`de`) instead of US English. See [`Base/res/keymaps/`](../Base/res/keymaps/) for a full list. Note that the `keymap` program itself will also modify the `/etc/Keyboard.ini` config file, but this way the change will persist across image rebuilds.
+
+## CMake build options
+
+There are some optional features that can be enabled during compilation that are intended to help with specific types of development work or introduce experimental features. Currently, the following build options are available:
+- `ENABLE_ADDRESS_SANITIZER` and `ENABLE_KERNEL_ADDRESS_SANITIZER`: builds in runtime checks for memory corruption bugs (like buffer overflows and memory leaks) in Lagom test cases and the kernel, respectively.
+- `ENABLE_MEMORY_SANITIZER`: enables runtime checks for uninitialized memory accesses in Lagom test cases.
+- `ENABLE_UNDEFINED_SANITIZER`: builds in runtime checks for [undefined behavior](https://en.wikipedia.org/wiki/Undefined_behavior) (like null pointer dereferences and signed integer overflows) in Lagom test cases.
```

Comment:
oop, this option enables UBSAN for Serenity Userland as well.

Reply:
Not sure where, but somewhere we'll want to mention that SERENITY_RAM_SIZE=1G or more is essentially required if running in graphical mode with UBSAN enabled. 512 MiB just won't cut it.

Reply:
I'm gonna leave this as-is for this PR, but yeah, we should improve this. :)